Data/pubtator_to_xml.py

```diff
@@ -0,0 +1,152 @@
+
+    Keyword Arguments:
+    file_object -- the file object used for writing
+    tagged_term -- look up
+    index -- keep track of the annotation id
+    """
+    file_object.write("<annotation id=\"%d\">\n" % (index))
+    file_object.write("<infon key=\"title\">%s</infon>\n" % (tagged_term[4]))
+    file_object.write("<location length=\"%d\" offset=\"%d\"></location>\n" % (len(tagged_term[3]), tagged_term[1]))
+    file_object.write("<text>%s</text>\n" % (tagged_term[3]))
+
+    # If the annotation type is a Gene,Species, Mutation, SNP
+    # Write out relevant tag
+    if tagged_term[4] == "Gene":
+        file_object.write("<infon key=\"NCBI Gene\">%s</infon>\n" % (tagged_term[5][0:tagged_term[5].find("(")]))
+
+    elif tagged_term[4] == "Species":
+        file_object.write("<infon key=\"NCBI Species\">%s</infon>\n" % (tagged_term[5]))
+
+    elif "Mutation" in tagged_term[4]:
+        file_object.write("<infon key=\"tmVar\">%s</infon>\n" % (tagged_term[5]))
+
+    elif "SNP" in tagged_term[4]:
+        file_object.write("<infon key=\"tmVar\">%s</infon>\n" % (tagged_term[5]))
+
+    else:
+        # If there is no MESH ID for an annotation
+        if len(tagged_term) > 5:
+            # check to see if there are multiple mesh tags
+            if "|" in tagged_term[5]:
+                # Write out each MESH id as own tag
+                for ids in tagged_term[5].split("|"):
+                    # Some ids dont have the MESH:#### form so added case to that
+                    if ":" not in ids:
+                        file_object.write("<infon key=\"MESH\">%s</infon>\n" % (tagged_term[5]))
+                    else:
+                        term_type, term_id = ids.split(":")
+                        file_object.write("<infon key=\"%s\">%s</infon>\n" % (term_type, term_id))
+            else:
+                # Some ids dont have the MESH:#### form so added case to that
+                if ":" in tagged_term[5]:
+                    term_type, term_id = tagged_term[5].split(":")
+                    file_object.write("<infon key=\"%s\">%s</infon>\n" % (term_type, term_id))
+                else:
+                    file_object.write("<infon key=\"MESH\">%s</infon>\n" % (tagged_term[5]))
+        else:
+            file_object.write("<infon key=\"MESH\">Unknown</infon>\n")
+    file_object.write("</annotation>\n")
+
+
+def convert_pubtator(input_file, output_file):
+    """Convert pubtators annotation list to XML
+
+    Keyword Arguments:
+    input_file -- the path of pubtators annotation file
+    output_file -- the path to output the converted text
+    """
+    term_tags = set()
+    with open(output_file, 'w') as g:
+        # Header for pubtator's xml format
+        g.write("<collection>\n<source>Pubtator</source>\n")
+        # read from gunzip file
+        with gzip.open(input_file, "rb") as f:
+
+            for line in tqdm.tqdm(f):
+                # Convert "illegal chracters" (i.e. < > &) in the main text
```

Comment:
What defines these as illegal characters? I'm hoping there is an automated encoder/decoder, so we don't have to hardcode these conversions.